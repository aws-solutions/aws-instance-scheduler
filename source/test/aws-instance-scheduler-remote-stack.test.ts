#!/usr/bin/env node
/*****************************************************************************
 *  Copyright 2020 Amazon.com, Inc. or its affiliates. All Rights Reserved.   *
 *                                                                            *
 *  Licensed under the Apache License, Version 2.0 (the "License"). You may   *
 *  not use this file except in compliance with the License. A copy of the    *
 *  License is located at                                                     *
 *                                                                            *
 *      http://www.apache.org/licenses/LICENSE-2.0                            *
 *                                                                            *
 *  or in the 'license' file accompanying this file. This file is distributed *
 *  on an 'AS IS' BASIS, WITHOUT WARRANTIES OR CONDITIONS OF ANY KIND,        *
 *  express or implied. See the License for the specific language governing   *
 *  permissions and limitations under the License.                            *
 *****************************************************************************/

import { SynthUtils } from '@aws-cdk/assert';
import * as cdk from '@aws-cdk/core';
import instanceSchedulerStack = require('../lib/aws-instance-scheduler-remote-stack');
import '@aws-cdk/assert/jest';


function getTestStack(): cdk.Stack {
  const envEU = { account: '111111111111', region: 'eu-west-1' };
  const app = new cdk.App();
  const stack = new instanceSchedulerStack.AwsInstanceSchedulerRemoteStack(app, 'stack', {env: envEU})

  const instanceSchedulerRemoteStackMetadata =
  {
    "AWS::CloudFormation::Interface": {
      "ParameterGroups": [{
        "Label": {
          "default": "Account"
        },
        "Parameters": [
          "InstanceSchedulerAccount"
        ]
      }],
      "ParameterLabels": {
        "InstanceSchedulerAccount": {
          "default": "Primary account"
        }
      }
    }
  }
  stack.templateOptions.metadata = instanceSchedulerRemoteStackMetadata
  stack.templateOptions.templateFormatVersion = "2010-09-09"
  return stack;
}
/*
 * SnapShot Testing for the AwsInstanceSchedulerStack.
 */
test('AwsInstanceSchedulerStack snapshot test', () => {
  expect(SynthUtils.toCloudFormation(getTestStack())).toMatchSnapshot();
});