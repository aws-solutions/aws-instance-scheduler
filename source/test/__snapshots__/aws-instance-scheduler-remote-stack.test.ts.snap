// Jest Snapshot v1, https://goo.gl/fbAQLP

exports[`AwsInstanceSchedulerStack snapshot test 1`] = `
Object {
  "AWSTemplateFormatVersion": "2010-09-09",
  "Metadata": Object {
    "AWS::CloudFormation::Interface": Object {
      "ParameterGroups": Array [
        Object {
          "Label": Object {
            "default": "Account",
          },
          "Parameters": Array [
            "InstanceSchedulerAccount",
          ],
        },
      ],
      "ParameterLabels": Object {
        "InstanceSchedulerAccount": Object {
          "default": "Primary account",
        },
      },
    },
  },
  "Outputs": Object {
    "CrossAccountRole": Object {
      "Description": "Arn for cross account role for Instance scheduler, add this arn to the list of crossaccount roles (CrossAccountRoles) parameter of the Instance Scheduler template.",
      "Value": Object {
        "Fn::GetAtt": Array [
          "EC2SchedulerCrossAccountRole",
          "Arn",
        ],
      },
    },
  },
  "Parameters": Object {
    "InstanceSchedulerAccount": Object {
      "AllowedPattern": "(^[0-9]{12}$)",
      "ConstraintDescription": "Account number is a 12 digit number",
      "Description": "Account number of Instance Scheduler account to give access to manage EC2 and RDS  Instances in this account.",
      "Type": "String",
    },
  },
  "Resources": Object {
    "EC2SchedulerCrossAccountRole": Object {
      "Metadata": Object {
        "cfn_nag": Object {
          "rules_to_suppress": Array [
            Object {
              "id": "W11",
              "reason": "All policies have been scoped to be as restrictive as possible. This solution needs to access ec2/rds resources across all regions.",
            },
          ],
        },
      },
      "Properties": Object {
        "AssumeRolePolicyDocument": Object {
          "Statement": Array [
            Object {
              "Action": "sts:AssumeRole",
              "Effect": "Allow",
              "Principal": Object {
                "AWS": Object {
                  "Fn::Join": Array [
                    "",
                    Array [
                      "arn:aws:iam::",
                      Object {
                        "Ref": "InstanceSchedulerAccount",
                      },
                      ":root",
                    ],
                  ],
                },
                "Service": "lambda.amazonaws.com",
              },
            },
          ],
          "Version": "2012-10-17",
        },
        "Path": "/",
        "Policies": Array [
          Object {
            "PolicyDocument": Object {
              "Statement": Array [
                Object {
                  "Action": Array [
                    "rds:DeleteDBSnapshot",
                    "rds:DescribeDBSnapshots",
                    "rds:StopDBInstance",
                  ],
                  "Effect": "Allow",
                  "Resource": "arn:aws:rds:*:111111111111:snapshot:*",
                },
                Object {
                  "Action": Array [
                    "rds:AddTagsToResource",
                    "rds:RemoveTagsFromResource",
                    "rds:DescribeDBSnapshots",
                    "rds:StartDBInstance",
                    "rds:StopDBInstance",
                  ],
                  "Effect": "Allow",
                  "Resource": "arn:aws:rds:*:111111111111:db:*",
                },
                Object {
                  "Action": Array [
                    "rds:AddTagsToResource",
                    "rds:RemoveTagsFromResource",
                    "rds:StartDBCluster",
                    "rds:StopDBCluster",
                  ],
                  "Effect": "Allow",
                  "Resource": "arn:aws:rds:*:111111111111:cluster:*",
                },
                Object {
                  "Action": Array [
                    "ec2:StartInstances",
                    "ec2:StopInstances",
                    "ec2:CreateTags",
                    "ec2:DeleteTags",
                  ],
                  "Effect": "Allow",
                  "Resource": "arn:aws:ec2:*:111111111111:instance/*",
                },
                Object {
                  "Action": Array [
                    "rds:DescribeDBClusters",
                    "rds:DescribeDBInstances",
                    "ec2:DescribeInstances",
                    "ec2:DescribeRegions",
                    "ec2:ModifyInstanceAttribute",
                    "ssm:DescribeMaintenanceWindows",
                    "ssm:DescribeMaintenanceWindowExecutions",
                    "tag:GetResources",
                  ],
                  "Effect": "Allow",
                  "Resource": "*",
                },
              ],
              "Version": "2012-10-17",
            },
            "PolicyName": "EC2InstanceSchedulerRemote",
          },
        ],
      },
      "Type": "AWS::IAM::Role",
    },
  },
}
`;
